# Исследование Casino players cashout

Данные взяты из "Metabase",таблица "Affiliate report". Интервал апрель-сентябрь 2020. Действия игроков. Предобработка на SQL не выполнялась. Нужно предобработать и исследовать данные, дать описание игрока который делает кэшаут и не делает для отдела продаж приложив статистику.

## Оглавление:

1. [Загрузка и просмотр данных](#chapter1)
2. [Предобработка и просмотр данных](#chapter2)<br>
3. [Описание Данных](#chapter3)
4. [Группировка и категоризация данных](#chapter4)
5. [Исследование данных](#chapter5)
6. [Общий вывод](#chapter6)

### 1. Загрузка и просмотр данных
<a class="anchor" id="chapter1"></a>

In [1]:
import pandas as pd
from datetime import datetime
import numpy as np
import io
import matplotlib.pyplot as plt

In [16]:
pd.set_option('display.max_rows', 100)
pd.set_option('display.max_columns', 30)
print(pd.get_option("display.max_rows"))
print(pd.get_option("display.max_columns"))

100
30


In [48]:
df = pd.read_csv('datasets/au_apr_sep_aff.csv')

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 135096 entries, 0 to 135095
Data columns (total 27 columns):
ID                           135096 non-null int64
Hash                         135096 non-null object
Date At                      135096 non-null object
Partner ID                   135096 non-null int64
Partner Name                 135095 non-null object
Brand ID                     135096 non-null int64
Brand Name                   135096 non-null object
Strategy                     135096 non-null object
Campaign ID                  135096 non-null int64
Campaign Name                135096 non-null object
Player Country               135096 non-null object
Registrations Count          135096 non-null int64
Currency                     86048 non-null object
First Deposits Count         86048 non-null float64
First Deposits Sum           86048 non-null float64
Deposits Count               86048 non-null float64
Deposits Sum                 86048 non-null float64
Cash Outs C

In [61]:
df

ID              Date At  Partner ID  \
0       100508071706619019  2020-09-17T03:00:00       84181   
1       100508071706619032  2020-09-17T03:00:00       84181   
2       100508071706619035  2020-09-17T03:00:00       84181   
3       100508071706619043  2020-09-17T03:00:00       84181   
4       100508071707473163  2020-09-17T03:00:00       84726   
...                    ...                  ...         ...   
135091  991408124614190987  2020-09-17T03:00:00       86027   
135092  991408124615645128  2020-09-17T03:00:00       86056   
135093  991408124622813177  2020-09-17T03:00:00       86522   
135094  991408124638551921  2020-09-17T03:00:00       87199   
135095  991408124639267754  2020-09-17T03:00:00       87199   

                  Partner Name  Brand ID Brand Name    Strategy  Campaign ID  \
0           WryReliableComalny        99   Getslots     CPA 110        99071   
1           WryReliableComalny        99   Getslots     CPA 130       101289   
2           WryReliableComalny        99   Getslots     CPA 130       101289   
3           WryReliableComalny        99   Getslots  CPA iOS 06        93298   
4       Silver Maple Solutions        99   Getslots         NGR        91512   
...                        ...       ...        ...         ...          ...   
135091                 AcidBro        99   Getslots     CPA 110        98235   
135092            SpicyCasinos        99   Getslots         NGR        94509   
135093                 CalMezh        99   Getslots  CPA iOS 06        95900   
135094       MS Experts s.r.o.        99   Getslots         NGR        96647   
135095       MS Experts s.r.o.        99   Getslots         NGR        96647   

                Campaign Name Player Country  Registrations Count Currency  \
0          Getslots AU AN ORG             AU                    1      NaN   
1       Getslots AU 130 I org             AU                    1      NaN   
2       Getslots AU 130 I org             AU                    1      NaN   
3               GSLOTS AU NEW             AU                    1      NaN   
4                        MAMA             AU                    0      AUD   
...                       ...            ...                  ...      ...   
135091           Getslots app             AU                    0      AUD   
135092            Default NGR             AU                    0      AUD   
135093           Getslots 120             AU                    0      AUD   
135094            Default NGR             AU                    1      AUD   
135095            Default NGR             AU                    1      AUD   

        First Deposits Count  First Deposits Sum  Deposits Count  \
0                        NaN                 NaN             NaN   
1                        NaN                 NaN             NaN   
2                        NaN                 NaN             NaN   
3                        NaN                 NaN             NaN   
4                        0.0                 0.0             1.0   
...                      ...                 ...             ...   
135091                   0.0                 0.0             1.0   
135092                   0.0                 0.0             0.0   
135093                   0.0                 0.0             1.0   
135094                   0.0                 0.0             0.0   
135095                   0.0                 0.0             0.0   

        Deposits Sum  Cash Outs Count  Cash Outs Sum  Bets Sum  Wins Sum  \
0                NaN              NaN            NaN       NaN       NaN   
1                NaN              NaN            NaN       NaN       NaN   
2                NaN              NaN            NaN       NaN       NaN   
3                NaN              NaN            NaN       NaN       NaN   
4               20.0              0.0            0.0     59.14     24.14   
...              ...              ...            ...       ...       ...   
135091          25.0

### 2. Предобработка и просмотр данных 
<a class="anchor" id="chapter2"></a>

In [51]:
df.drop(columns=['Hash'], inplace=True)

In [37]:
df['Player ID'].value_counts()

86048

Посмотрим на уников плееров с NaN , 1 ли ID или бывает несколько строк на 1 плеера

In [64]:
df.query('`Real Ngr` == "NaN"').sort_values(by='Date At')['Player ID'].value_counts().mean()

1.0

In [67]:
df.query('`Real Ngr` == "NaN"')['Player ID'].count()

49048

По средней видим 1 плеер ID с NaN 1 строка, полагаю это игроки без депов и активности, их удалим, таких 50 000 из 135 000

In [68]:
df.dropna(inplace=True)

Посмотрим на нашу таблицу с кешаут больше нуля и отсортируем ее по плеерам и месяцам

In [233]:
df.query('`Player ID` == 3857272').sort_values(by='Date At')

ID              Date At  Partner ID Partner Name  \
88529   386207795030518944  2020-04-16T03:00:00       82464    Geek Team   
88650   386407795030442163  2020-04-18T03:00:00       82464    Geek Team   
69091   265877798403011787  2020-05-26T03:00:00       82464    Geek Team   
71032   265907798402768269  2020-05-29T03:00:00       82464    Geek Team   
71585   265917798403848528  2020-05-30T03:00:00       82464    Geek Team   
15109   200917792755700230  2020-06-08T03:00:00       82464    Geek Team   
15807   200927792755539162  2020-06-09T03:00:00       82464    Geek Team   
24338   213467788326145341  2020-07-08T03:00:00       82464    Geek Team   
28030   213507788325430546  2020-07-12T03:00:00       82464    Geek Team   
29278   213517788324723303  2020-07-13T03:00:00       82464    Geek Team   
30445   213527788325326887  2020-07-14T03:00:00       82464    Geek Team   
31932   213537788325546363  2020-07-15T03:00:00       82464    Geek Team   
41352   242277788510887073  2020-07-16T03:00:00       82464    Geek Team   
43682   242297788506914968  2020-07-18T03:00:00       82464    Geek Team   
47154   242327788509855896  2020-07-21T03:00:00       82464    Geek Team   
49804   242347788508647437  2020-07-23T03:00:00       82464    Geek Team   
52082   242367788510889247  2020-07-25T03:00:00       82464    Geek Team   
55189   242397788510719210  2020-07-28T03:00:00       82464    Geek Team   
102406  696007786511875760  2020-08-11T03:00:00       82464    Geek Team   
104944  696207786511872191  2020-08-13T03:00:00       82464    Geek Team   
107747  696407786512787526  2020-08-15T03:00:00       82464    Geek Team   
118637  697207786512685490  2020-08-23T03:00:00       82464    Geek Team   
121125  697407786510520524  2020-08-25T03:00:00       82464    Geek Team   
126582  697807786511567956  2020-08-29T03:00:00       82464    Geek Team   
127857  697907786510532213  2020-08-30T03:00:00       82464    Geek Team   
78797   299367785308257835  2020-09-05T03:00:00       82464    Geek Team   

        Brand ID Brand Name Strategy  Campaign ID Campaign Name  \
88529         99   Getslots      CPA        87195      Getslots   
88650         99   Getslots      CPA        87195      Getslots   
69091         99   Getslots      CPA        87195      Getslots   
71032         99   Getslots      CPA        87195      Getslots   
71585         99   Getslots      CPA        87195      Getslots   
15109         99   Getslots      CPA        87195      Getslots   
15807         99   Getslots      CPA        87195      Getslots   
24338         99   Getslots      CPA        87195      Getslots   
28030         99   Getslots      CPA        87195      Getslots   
29278         99   Getslots      CPA        87195      Getslots   
30445         99   Getslots      CPA        87195      Getslots   
31932         99   Getslots      CPA        87195      Getslots   
41352         99   Getslots      CPA        87195      Getslots   
43682         99   Getslots      CPA        87195      Getslots   
47154         99   Getslots      CPA        87195      Getslots   
49804         99   Getslots      CPA        87195      Getslots   
52082         99   Getslots      CPA        87195      Getslots   
55189         99   Getslots      CPA        87195      Getslots   
102406        99   Getslots      CPA        87195      Getslots   
104944        99   Getslots      CPA        87195      Getslots   
107747        99   Getslots      CPA        87195      Getslots   
118637        99   Getslots      CPA        87195      Getslots   
121125        99   Getslots      CPA        87195      Getslots   
126582        99   Getslots      CPA        87195      Getslots   
127857        99   Getslots      CPA        87195      Getslots   
78797         99   Getslots      CPA        87195      Getslots   

       Player Country  Registrations Count Currency  First Deposits Count  \
88529              AU                    1      AUD                   1.0   
88650  

In [135]:
(df.sort_values(by=['Player ID','Date At'])
)

ID              Date At  Partner ID  \
88324  356307794999171454  2020-04-15T03:00:00       82464   
88326  356307794999171469  2020-04-15T03:00:00       82464   
88328  356307794999171481  2020-04-15T03:00:00       82464   
88331  356307794999171488  2020-04-15T03:00:00       82464   
88333  356307794999171493  2020-04-15T03:00:00       82464   
...                   ...                  ...         ...   
8999   181299305381806214  2020-09-30T03:00:00       82469   
93717  666709309176030624  2020-09-30T03:00:00       84181   
93730  666709309186195615  2020-09-30T03:00:00       85405   
93738  666709309238476330  2020-09-30T03:00:00       88081   
93732  666709309186195620  2020-09-30T03:00:00       85405   

                     Partner Name  Brand ID Brand Name Strategy  Campaign ID  \
88324                   Geek Team        99   Getslots      CPA        87195   
88326                   Geek Team        99   Getslots      CPA        87195   
88328                   Geek Team        99   Getslots      CPA        87195   
88331                   Geek Team        99   Getslots      CPA        87195   
88333                   Geek Team        99   Getslots      CPA        87195   
...                           ...       ...        ...      ...          ...   
8999                   Union Team        99   Getslots   CPA 07        98363   
93717          WryReliableComalny        99   Getslots  CPA 009       100473   
93730                 Volkov Team        99   Getslots  CPA 009       100465   
93738  Carrington Oliver John ltd        99   Getslots      NGR        98702   
93732                 Volkov Team        99   Getslots  CPA 009       100465   

               Campaign Name Player Country  Registrations Count Currency  \
88324               Getslots             AU                    1      AUD   
88326               Getslots             AU                    1      AUD   
88328               Getslots             AU                    1      AUD   
88331               Getslots             AU                    1      AUD   
88333               Getslots             AU                    1      AUD   
...                      ...            ...                  ...      ...   
8999        Getslots web 100             AU                    1      AUD   
93717  Getslots AU 50 org an             AU                    1      AUD   
93730  Getslots 50 AU 603 an             AU                    1      AUD   
93738            Default NGR             AU                    1      AUD   
93732  Getslots 50 AU 603 an             AU                    1      AUD   

       First Deposits Count  First Deposits Sum  Deposits Count  Deposits Sum  \
88324                   1.0                20.0             1.0          20.0   
88326                   1.0                20.0             2.0          40.0   
88328                   1.0                20.0             1.0          20.0   
88331                   1.0               100.0             1.0         100.0   
88333                   1.0                15.0             1.0          15.0   
...                     ...                 ...             ...           ...   
8999                    1.0                50.0             2.0         100.0   
93717                   1.0                50.0             3.0         100.0   
93730                   1.0                25.0             1.0          25.0   
93738                   0.0                 0.0             0.0           0.0   
93732                   1.0                25.0             1.0          25.0   

       Cash Outs Count  Cash Outs Sum  Bets Sum  Wins Sum  Bonus Issues Sum  \
88324              0.0            0.0     75.80     56.15              0.00   
88326              0.0            0.0      0.00      0.00              0.00   
88328              0.0            0.0     20.50      0.50              0.00   
88331              0.0            0.0    275.00     75.00            100.00   
88333              6.0  

### 3. Описание Данных
<a class="anchor" id="chapter3"></a>

Опишем наши данные

In [246]:
df.query('`Player ID` == 5129537').sort_values(by='Date At')


ID              Date At  Partner ID Partner Name  \
82847  299407785311216888  2020-09-09T03:00:00       82464    Geek Team   
83893  299417785311951514  2020-09-10T03:00:00       82464    Geek Team   
34333  234088147268653025  2020-09-18T03:00:00       82464    Geek Team   
83     104048678743650447  2020-09-24T03:00:00       82464    Geek Team   
88270  349009052456773653  2020-09-28T03:00:00       82464    Geek Team   
88417  364709154378092399  2020-09-29T03:00:00       82464    Geek Team   
73284  272499248749826316  2020-09-30T03:00:00       82464    Geek Team   

       Brand ID Brand Name Strategy  Campaign ID    Campaign Name  \
82847        99   Getslots  CPA 110        97540  Getslots 110 AU   
83893        99   Getslots  CPA 110        97540  Getslots 110 AU   
34333        99   Getslots  CPA 110        97540  Getslots 110 AU   
83           99   Getslots  CPA 110        97540  Getslots 110 AU   
88270        99   Getslots  CPA 110        97540  Getslots 110 AU   
88417        99   Getslots  CPA 110        97540  Getslots 110 AU   
73284        99   Getslots  CPA 110        97540  Getslots 110 AU   

      Player Country  Registrations Count Currency  First Deposits Count  \
82847             AU                    0      AUD                   1.0   
83893             AU                    0      AUD                   0.0   
34333             AU                    0      AUD                   0.0   
83                AU                    0      AUD                   0.0   
88270             AU                    0      AUD                   0.0   
88417             AU                    0      AUD                   0.0   
73284             AU                    0      AUD                   0.0   

       First Deposits Sum  Deposits Count  Deposits Sum  Cash Outs Count  \
82847                25.0             1.0          25.0              0.0   
83893                 0.0             0.0           0.0              0.0   
34333                 0.0             0.0           0.0              0.0   
83                    0.0             0.0           0.0              0.0   
88270                 0.0             0.0           0.0              0.0   
88417                 0.0             0.0           0.0              0.0   
73284                 0.0             0.0           0.0              0.0   

       Cash Outs Sum  Bets Sum  Wins Sum  Bonus Issues Sum  \
82847            0.0    275.95    353.55              25.0   
83893            0.0     79.00     24.40               0.0   
34333            0.0      3.50      0.60               0.0   
83               0.0      0.00      0.00             -45.1   
88270            0.0      5.50      0.00               0.0   
88417            0.0     29.50     30.70               0.0   
73284            0.0     21.00      1.25               0.0   

       Additional Deductions Sum  Real Ngr           Created At  \
82847                        0.0   -102.60  2020-09-14T08:19:19   
83893                        0.0     54.60  2020-09-14T08:19:20   
34333                        0.0      2.90  2020-09-18T08:12:29   
83                           0.0     45.10  2020-09-24T05:00:03   
88270                        0.0      5.50  2020-09-28T08:00:03   
88417                        0.0     -1.20  2020-09-29T11:00:03   
73284                        0.0     19.75  2020-09-30T12:00:03   

                Updated At  Player ID  
82847  2020-09-14T08:19:19    5129537  
83893  2020-09-14T08:19:20    5129537  
34333  2020-09-19T23:59:49    5129537  
83     2020-10-05T14:35:22    5129537  
88270  2020-10-05T14:35:22    5129537  
88417  2020-10-05T14:35:22    5129537  
73284  2020-10-05T14:35:22    5129537

In [185]:
df_filter = df[['Deposits Count', 'Deposits Sum', 'Cash Outs Count', 'Cash Outs Sum','Bets Sum','Wins Sum','Bonus Issues Sum','Additional Deductions Sum']].describe()

Deposits Count  Deposits Sum  Cash Outs Count  Cash Outs Sum  \
count    86048.000000  86048.000000     86048.000000   86048.000000   
mean         1.337393     88.834844         0.386831      30.256112   
std          1.986803    399.134787         1.707837     617.948338   
min          0.000000      0.000000         0.000000  -23000.000000   
25%          0.000000      0.000000         0.000000       0.000000   
50%          1.000000     25.000000         0.000000       0.000000   
75%          2.000000     75.000000         0.000000       0.000000   
max         42.000000  55000.000000       164.000000   90000.000000   

            Bets Sum       Wins Sum  Bonus Issues Sum  \
count   86048.000000   86048.000000      86048.000000   
mean     1468.589428    1400.541004         13.565320   
std      8635.630669    8635.742781         73.233566   
min         0.000000       0.000000     -10468.800000   
25%        29.440000       9.700000          0.000000   
50%       175.000000     118.625000          0.000000   
75%       759.697500     658.255000         18.735000   
max    784033.000000  780752.007451       2538.000000   

       Additional Deductions Sum  
count               86048.000000  
mean                   -0.725418  
std                    61.901045  
min                -10732.040000  
25%                     0.000000  
50%                     0.000000  
75%                     0.000000  
max                   620.670000

In [250]:
print("all string count =",df['Cash Outs Sum'].count())
print("all cashouts string sum =",df['Cash Outs Sum'].sum())
print('cashouts < 0 counts =',df.query('`Cash Outs Sum` < 0')['Cash Outs Sum'].count())
print('cashouts < 0 sum =',df.query('`Cash Outs Sum` < 0')['Cash Outs Sum'].sum())
print('cashouts > 0 counts =',df.query('`Cash Outs Sum` > 0')['Cash Outs Sum'].count())
print('cashouts > 0 sum =',df.query('`Cash Outs Sum` > 0')['Cash Outs Sum'].sum())

all string count = 86048
all cashouts string sum = 2603477.90011242
cashouts < 0 counts = 782
cashouts < 0 sum = -806653.4
cashouts > 0 counts = 3971
cashouts > 0 sum = 3410131.30011242


In [248]:
print('sum deps $ =',df['Deposits Sum'].sum())
print('sum cashout $ =',df['Cash Outs Sum'].sum())

sum deps $ = 7644060.682279689
sum cashout $ = 2603477.90011242


Посмотрим столбцы Additional Deductions Sum и Bonus Issues Sum

In [229]:
print('`Additional Deductions Sum` < 0 count =',df.query('`Additional Deductions Sum` < 0')['Additional Deductions Sum'].count())
print('`Additional Deductions Sum` < 0 sum =',df.query('`Additional Deductions Sum` < 0')['Additional Deductions Sum'].sum())
print('`Additional Deductions Sum` > 0 count =',df.query('`Additional Deductions Sum` > 0')['Additional Deductions Sum'].count())
print('`Additional Deductions Sum` > 0 sum =',df.query('`Additional Deductions Sum` > 0')['Additional Deductions Sum'].sum())

`Additional Deductions Sum` < 0 count = 54
`Additional Deductions Sum` < 0 sum = -63196.33
`Additional Deductions Sum` > 0 count = 3
`Additional Deductions Sum` > 0 sum = 775.6


In [231]:
print('`Bonus Issues Sum` < 0 count =',df.query('`Bonus Issues Sum` < 0')['Bonus Issues Sum'].count())
print('`Bonus Issues Sum` < 0 sum =',df.query('`Bonus Issues Sum` < 0')['Bonus Issues Sum'].sum())
print('`Bonus Issues Sum` > 0 count =',df.query('`Bonus Issues Sum` > 0')['Bonus Issues Sum'].count())
print('`Bonus Issues Sum` > 0 sum =',df.query('`Bonus Issues Sum` > 0')['Bonus Issues Sum'].sum())

`Bonus Issues Sum` < 0 count = 9807
`Bonus Issues Sum` < 0 sum = -206873.61004134
`Bonus Issues Sum` > 0 count = 32234
`Bonus Issues Sum` > 0 sum = 1374142.2368931598


In [241]:
df.shape

(86048, 26)

### Вывод

Мы видим столюцы с депами, выводами , бонусами, датами и ID по каждому игроку без группировки. У нас минусовые, значения в Кэшауте, это отмененные выводы, итоговая сумма по тоталу берется.
Additional Deductions Sum - столбец с доп отчислениями и ручными корректировками, он имеет менне процента от валовой выручки его удаляем.
Bonus Issues Sum - есть минусовые бонусы и плюсовые, плюсовые имеют большой тотал 1,4 млн, около 20% от валовой выручки.

### 4. Группировка данных и категоризация
<a class="anchor" id="chapter4"></a>

Посмотрим на типового игрока который делает, кэшаут

In [265]:
df.query('`Player ID` == 3900150').sort_values(by='Date At')

ID              Date At  Partner ID Partner Name  \
89140   386707795062899180  2020-04-21T03:00:00       82469   Union Team   
89257   386807795039591642  2020-04-22T03:00:00       82469   Union Team   
89431   386907795040446467  2020-04-23T03:00:00       82469   Union Team   
90755   387607795040645406  2020-04-30T03:00:00       82469   Union Team   
64477   265797798409096820  2020-05-18T03:00:00       82469   Union Team   
64966   265807798412205913  2020-05-19T03:00:00       82469   Union Team   
72669   265927798513520939  2020-05-31T03:00:00       82469   Union Team   
10430   200847792857222622  2020-06-01T03:00:00       82469   Union Team   
14018   200897792856538040  2020-06-06T03:00:00       82469   Union Team   
16803   200937792870846528  2020-06-10T03:00:00       82469   Union Team   
20549   200987792850328602  2020-06-15T03:00:00       82469   Union Team   
54202   242387788650890316  2020-07-27T03:00:00       82469   Union Team   
59027   242427788585219348  2020-07-31T03:00:00       82469   Union Team   
91598   519007789926794982  2020-08-01T03:00:00       82469   Union Team   
106590  696307786568101885  2020-08-14T03:00:00       82469   Union Team   
85935   299437785317101047  2020-09-12T03:00:00       82469   Union Team   
60441   249467784754188600  2020-09-13T03:00:00       82469   Union Team   
130218  703408169988425324  2020-09-18T03:00:00       82469   Union Team   

        Brand ID Brand Name Strategy  Campaign ID Campaign Name  \
89140         99   Getslots      CPA        87197      Getslots   
89257         99   Getslots      CPA        87197      Getslots   
89431         99   Getslots      CPA        87197      Getslots   
90755         99   Getslots      CPA        87197      Getslots   
64477         99   Getslots      CPA        87197      Getslots   
64966         99   Getslots      CPA        87197      Getslots   
72669         99   Getslots      CPA        87197      Getslots   
10430         99   Getslots      CPA        87197      Getslots   
14018         99   Getslots      CPA        87197      Getslots   
16803         99   Getslots      CPA        87197      Getslots   
20549         99   Getslots      CPA        87197      Getslots   
54202         99   Getslots      CPA        87197      Getslots   
59027         99   Getslots      CPA        87197      Getslots   
91598         99   Getslots      CPA        87197      Getslots   
106590        99   Getslots      CPA        87197      Getslots   
85935         99   Getslots      CPA        87197      Getslots   
60441         99   Getslots      CPA        87197      Getslots   
130218        99   Getslots      CPA        87197      Getslots   

       Player Country  Registrations Count Currency  First Deposits Count  \
89140              AU                    1      AUD                   1.0   
89257              AU                    0      AUD                   0.0   
89431              AU                    0      AUD                   0.0   
90755              AU                    0      AUD                   0.0   
64477              AU                    0      AUD                   0.0   
64966              AU                    0      AUD                   0.0   
72669              AU                    0      AUD                   0.0   
10430              AU                    0      AUD                   0.0   
14018              AU                    0      AUD                   0.0   
16803              AU                    0      AUD                   0.0   
20549              AU                    0      AUD                   0.0   
54202              AU                    0      AUD                   0.0   
59027              AU                    0      AUD                   0.0   
91598              AU                    0      AUD                   0.0   
106590             AU                    0      AUD                   0.0   
85935              AU                    0      AUD                   0.0

Сведем фильтрованные данные которые нам интересны, в отдельный df_filter 

In [410]:
df_filter = df[['Date At','Player ID','First Deposits Sum','Deposits Count', 'Deposits Sum', 'Cash Outs Count', 'Cash Outs Sum','Bets Sum','Wins Sum','Bonus Issues Sum']].copy()
df_filter

Date At  Player ID  First Deposits Sum  Deposits Count  \
4       2020-09-17T03:00:00    4554390                 0.0             1.0   
5       2020-09-17T03:00:00    5004487                 0.0             0.0   
7       2020-09-17T03:00:00    5504921                50.0             1.0   
9       2020-09-17T03:00:00    5504961                50.0             1.0   
10      2020-09-17T03:00:00    5504953                50.0             3.0   
...                     ...        ...                 ...             ...   
135091  2020-09-17T03:00:00    5247918                 0.0             1.0   
135092  2020-09-17T03:00:00    5500981                 0.0             0.0   
135093  2020-09-17T03:00:00    5134536                 0.0             1.0   
135094  2020-09-17T03:00:00    5511007                 0.0             0.0   
135095  2020-09-17T03:00:00    5511462                 0.0             0.0   

        Deposits Sum  Cash Outs Count  Cash Outs Sum  Bets Sum  Wins Sum  \
4               20.0              0.0            0.0     59.14     24.14   
5                0.0              0.0            0.0    143.95     93.70   
7               50.0              0.0            0.0    181.30     81.50   
9               50.0              0.0            0.0    195.00     95.20   
10             200.0              0.0            0.0   2682.20   2316.88   
...              ...              ...            ...       ...       ...   
135091          25.0              0.0            0.0      8.00      0.88   
135092           0.0              0.0            0.0      0.00      0.30   
135093          15.0              0.0            0.0     86.75     71.80   
135094           0.0              0.0            0.0      2.50      0.15   
135095           0.0              0.0            0.0     10.50      5.35   

        Bonus Issues Sum  
4                  15.00  
5                   0.00  
7                  50.00  
9                  50.00  
10                166.25  
...                  ...  
135091              0.00  
135092              0.00  
135093              0.00  
135094              2.45  
135095              5.25  

[86048 rows x 10 columns]

Из df_filter оставим строки с днями, когда игрок играл был активен, либо депал или выводил либо играл

In [411]:
df_filter_act = df_filter.query('`Deposits Sum` != 0 or `Cash Outs Sum` != 0 or `Bets Sum` > 0').copy()

In [412]:
df_filter_act.reset_index(inplace=True,drop=True)
df_filter_act

Date At  Player ID  First Deposits Sum  Deposits Count  \
0      2020-09-17T03:00:00    4554390                 0.0             1.0   
1      2020-09-17T03:00:00    5004487                 0.0             0.0   
2      2020-09-17T03:00:00    5504921                50.0             1.0   
3      2020-09-17T03:00:00    5504961                50.0             1.0   
4      2020-09-17T03:00:00    5504953                50.0             3.0   
...                    ...        ...                 ...             ...   
74937  2020-09-17T03:00:00    5374244                 0.0             1.0   
74938  2020-09-17T03:00:00    5247918                 0.0             1.0   
74939  2020-09-17T03:00:00    5134536                 0.0             1.0   
74940  2020-09-17T03:00:00    5511007                 0.0             0.0   
74941  2020-09-17T03:00:00    5511462                 0.0             0.0   

       Deposits Sum  Cash Outs Count  Cash Outs Sum  Bets Sum  Wins Sum  \
0              20.0              0.0            0.0     59.14     24.14   
1               0.0              0.0            0.0    143.95     93.70   
2              50.0              0.0            0.0    181.30     81.50   
3              50.0              0.0            0.0    195.00     95.20   
4             200.0              0.0            0.0   2682.20   2316.88   
...             ...              ...            ...       ...       ...   
74937          70.0              0.0            0.0    414.25    292.10   
74938          25.0              0.0            0.0      8.00      0.88   
74939          15.0              0.0            0.0     86.75     71.80   
74940           0.0              0.0            0.0      2.50      0.15   
74941           0.0              0.0            0.0     10.50      5.35   

       Bonus Issues Sum  
0                 15.00  
1                  0.00  
2                 50.00  
3                 50.00  
4                166.25  
...                 ...  
74937             52.50  
74938              0.00  
74939              0.00  
74940              2.45  
74941              5.25  

[74942 rows x 10 columns]

### Вывод

Данные сгруппированы теперь можно их исследовать

### 5. Исследование данных
<a class="anchor" id="chapter5"></a>

#### Теперь использую наши вопросы

- Сколько плюсовых дней до кешаута?
- Какая обычная сумма (разница между wins - bets), предшествует кешауту? 
- Зависимость от суммы депозитов игрока (будут разные показатели для тех, кто депает много и мало).

Создадим столбец с кэшаут c ключом где событие вывода будет равно 1, а событие отмены вывода равно -1

In [480]:
df_filter_act['Cashout_key_plus'] = (df_filter_act['Cash Outs Sum'] > 0)*1
df_filter_act['Cashout_key_minus'] = (df_filter_act['Cash Outs Sum'] < 0)*(-1)

In [481]:
df_filter_act['Cashout_key'] = df_filter_act['Cashout_key_plus'] + df_filter_act['Cashout_key_minus'] 

In [482]:
df_filter_act.drop(columns=['Cashout_key_plus','Cashout_key_minus'], inplace=True)

In [483]:
df_filter_act.query('Cashout_key == -1')

Date At  Player ID  First Deposits Sum  Deposits Count  \
536    2020-05-03T03:00:00    3991790                 0.0             0.0   
547    2020-05-03T03:00:00    3916119                 0.0             0.0   
555    2020-05-03T03:00:00    3936280                 0.0             2.0   
580    2020-05-03T03:00:00    3985651                 0.0             0.0   
586    2020-05-03T03:00:00    3989914                 0.0             0.0   
...                    ...        ...                 ...             ...   
73408  2020-07-04T03:00:00    4553256                 0.0             0.0   
73459  2020-07-04T03:00:00    4547858                 0.0             0.0   
73590  2020-07-05T03:00:00    4548673                 0.0             0.0   
73681  2020-07-06T03:00:00    4559214                 0.0             0.0   
74448  2020-07-12T03:00:00    4567726                 0.0             7.0   

       Deposits Sum  Cash Outs Count  Cash Outs Sum  Bets Sum  Wins Sum  \
536             0.0             -1.0          -50.0    506.29    490.83   
547             0.0             -1.0        -3000.0    383.75     93.70   
555           100.0             -3.0         -100.0    276.24    276.10   
580             0.0             -1.0         -200.0    851.50    811.35   
586             0.0             -7.0         -150.0    253.90    203.92   
...             ...              ...            ...       ...       ...   
73408           0.0             -1.0         -180.0    618.00    434.65   
73459           0.0             -1.0         -150.0    197.00     47.28   
73590           0.0             -1.0         -900.0     35.00      6.50   
73681           0.0             -1.0         -850.0      0.00      0.00   
74448         875.0             -3.0         -800.0  10632.75   8907.25   

       Bonus Issues Sum  Cashout_key  income_player  
536                0.00           -1         -15.46  
547                0.00           -1        -290.05  
555                0.00           -1          -0.14  
580                0.00           -1         -40.15  
586                0.25           -1         -49.98  
...                 ...          ...            ...  
73408              0.00           -1        -183.35  
73459              0.00           -1        -149.72  
73590              0.00           -1         -28.50  
73681              0.00           -1           0.00  
74448             16.25           -1       -1725.50  

[782 rows x 12 columns]

Поменяем каунты по кэшаутам на минусовые, когда у нас была отмена списания

In [484]:
df_filter_act['Cash Outs Count'] = df_filter_act['Cash Outs Count'] * df_filter_act['Cashout_key']

Проверим статистику по клиенту который делал кэшаут

In [485]:
df_filter_act.query('`Player ID` == 3900150').sort_values(by='Date At')

Date At  Player ID  First Deposits Sum  Deposits Count  \
49491  2020-04-21T03:00:00    3900150               200.0             1.0   
49542  2020-04-22T03:00:00    3900150                 0.0             0.0   
49614  2020-04-23T03:00:00    3900150                 0.0             2.0   
50303  2020-04-30T03:00:00    3900150                 0.0             1.0   
35728  2020-05-18T03:00:00    3900150                 0.0             2.0   
35974  2020-05-19T03:00:00    3900150                 0.0             1.0   
40144  2020-05-31T03:00:00    3900150                 0.0             2.0   
5556   2020-06-01T03:00:00    3900150                 0.0             3.0   
7540   2020-06-06T03:00:00    3900150                 0.0             1.0   
9063   2020-06-10T03:00:00    3900150                 0.0             2.0   
11226  2020-06-15T03:00:00    3900150                 0.0             1.0   
30090  2020-07-27T03:00:00    3900150                 0.0             1.0   
32697  2020-07-31T03:00:00    3900150                 0.0             1.0   
50866  2020-08-01T03:00:00    3900150                 0.0             1.0   
59667  2020-08-14T03:00:00    3900150                 0.0             1.0   
47874  2020-09-12T03:00:00    3900150                 0.0             1.0   
72722  2020-09-18T03:00:00    3900150                 0.0             1.0   

       Deposits Sum  Cash Outs Count  Cash Outs Sum  Bets Sum  Wins Sum  \
49491         200.0              3.0         1000.0   6973.00   7623.00   
49542           0.0              8.0         2000.0  16050.00  18021.00   
49614         800.0              0.0            0.0   2715.00   1916.00   
50303         200.0              0.0            0.0    281.25     79.50   
35728         400.0              1.0          400.0   6788.90   6788.95   
35974         200.0              0.0            0.0    776.00    577.40   
40144         150.0              0.0            0.0   5169.50   5019.50   
5556          200.0              0.0            0.0    839.75    638.20   
7540           80.0              0.0            0.0    450.00    370.00   
9063           75.0              1.0          250.0    665.50    842.10   
11226         100.0              0.0            0.0    127.75     26.25   
30090         100.0              0.0            0.0    930.50    830.30   
32697         100.0              1.0          200.0    964.56   1064.75   
50866         100.0              0.0            0.0    320.50    220.50   
59667         100.0              0.0            0.0    197.20     97.00   
47874         150.0              0.0            0.0   1190.95   1041.00   
72722         100.0              0.0            0.0    143.20     40.20   

       Bonus Issues Sum  Cashout_key  income_player  
49491             150.0            1         650.00  
49542              30.0            1        1971.00  
49614               0.0            0        -799.00  
50303               0.0            0        -201.75  
35728               0.0            1           0.05  
35974               0.0            0        -198.60  
40144               0.0            0        -150.00  
5556                0.0            0        -201.55  
7540                0.0            0         -80.00  
9063                0.0            1         176.60  
11226               0.0            0        -101.50  
30090               0.0            0        -100.20  
32697               0.0            1         100.19  
50866               0.0            0        -100.00  
59667               0.0            0        -100.20  
47874               0.0            0        -149.95  
72722               0.0            0        -103.00

Посмотрим на распределение по количеству кешаутов при разных условиях `Bets Sum` <= `Wins Sum` и `Bets Sum` > `Wins Sum`

In [486]:
print('total cashout count =',df_filter_act.query('`Cashout_key` == 1')['Cashout_key'].sum())
print('total cashout sum =',df_filter_act.query('`Cashout_key` == 1')['Cash Outs Sum'].sum())


total cashout count = 3971
total cashout sum = 3410131.30011242


In [487]:
print('cashout `Bets Sum` <= `Wins Sum` count = ',df_filter_act.query('`Bets Sum` <= `Wins Sum` and `Cashout_key` == 1')['Cashout_key'].sum())
print('cashout `Bets Sum` <= `Wins Sum` sum =',df_filter_act.query('`Bets Sum` <= `Wins Sum` and `Cashout_key` == 1')['Cash Outs Sum'].sum())

cashout `Bets Sum` <= `Wins Sum` count =  2810
cashout `Bets Sum` <= `Wins Sum` sum = 2603604.13231116


In [488]:
print("cashout `Bets Sum` > `Wins Sum` count =",df_filter_act.query('`Bets Sum` > `Wins Sum` and `Cashout_key` == 1')['Cashout_key'].sum())
print("cashout `Bets Sum` > `Wins Sum` sum =",df_filter_act.query('`Bets Sum` > `Wins Sum` and `Cashout_key` == 1')['Cash Outs Sum'].sum())

cashout `Bets Sum` > `Wins Sum` count = 1161
cashout `Bets Sum` > `Wins Sum` sum = 806527.16780126


In [489]:
2603/3410

0.7633431085043988

#### Мы видим распределение по обороту денег 75% объема выводят в день когда выйгрыш больше или равен тратам, 25% выводят в день, когда выйгрыш меньше затрат игрока.

Создадим столбец "wins_key"с ключами по данным когда выйгрыш был больше или равен проигрышу и столбец с размером выйгрыша

In [490]:
df_filter_act['wins_key'] = (df_filter_act['Wins Sum'] >= df_filter_act['Bets Sum'])*1

In [491]:
df_filter_act['income_player'] = df_filter_act['Wins Sum'] - df_filter_act['Bets Sum']

In [492]:
df_filter_act.query('`Bets Sum` <= `Wins Sum`')

Date At  Player ID  First Deposits Sum  Deposits Count  \
5      2020-09-17T03:00:00    4558618                 0.0             0.0   
12     2020-09-17T03:00:00    4862849                25.0             1.0   
28     2020-09-14T03:00:00    4416610                 0.0             2.0   
30     2020-09-14T03:00:00    4624435                 0.0             0.0   
59     2020-09-24T03:00:00    4676921                 0.0             7.0   
...                    ...        ...                 ...             ...   
74880  2020-09-15T03:00:00    5492423                25.0             1.0   
74890  2020-09-17T03:00:00    5503455                25.0             1.0   
74893  2020-09-24T03:00:00    5487976                 0.0             3.0   
74905  2020-09-20T03:00:00    4994399                 0.0             1.0   
74932  2020-09-15T03:00:00    5214897                 0.0             1.0   

       Deposits Sum  Cash Outs Count  Cash Outs Sum  Bets Sum  Wins Sum  \
5               0.0              0.0            0.0    337.50    575.50   
12             25.0              1.0          150.0   7000.50   7121.55   
28            175.0              0.0            0.0    958.00    964.55   
30              0.0              0.0            0.0    146.00    184.90   
59            525.0              2.0         1600.0   5030.35   6006.04   
...             ...              ...            ...       ...       ...   
74880          25.0              0.0            0.0     94.00     94.65   
74890          25.0              0.0            0.0     94.50    120.70   
74893         150.0              0.0            0.0      0.00      0.00   
74905         100.0              0.0            0.0   3809.60   4170.70   
74932         125.0              0.0            0.0    457.50    731.50   

       Bonus Issues Sum  Cashout_key  income_player  wins_key  
5                  0.00            0         238.00         1  
12                25.00            1         121.05         1  
28                 0.00            0           6.55         1  
30                 0.00            0          38.90         1  
59                 0.00            1         975.69         1  
...                 ...          ...            ...       ...  
74880             25.00            0           0.65         1  
74890             30.22            0          26.20         1  
74893              0.00            0           0.00         1  
74905             40.00            0         361.10         1  
74932              0.00            0         274.00         1  

[11082 rows x 13 columns]

In [479]:
df_filter_act['Player ID'].value_counts().count()

21479

Соберем в отдельный отчет инфу по депам и кешаутам по каждому плееру

In [519]:
player_report = (df_filter_act
    .sort_values(by=['Player ID','Date At'])
    .groupby(['Player ID'])
    .agg({'First Deposits Sum':['sum'],'Deposits Count' : ['sum'],'Deposits Sum' : ['sum'],'Cash Outs Count' : ['sum'], 'Cash Outs Sum':['sum'],'Bets Sum':['sum'],'Wins Sum':['sum'],'Bonus Issues Sum':['sum'],'income_player':['sum'],'Cashout_key':['sum'],'wins_key':['sum']})
    .round(2)
    .reset_index()
)
player_report

Player ID First Deposits Sum Deposits Count Deposits Sum  \
                               sum            sum          sum   
0       3845721               20.0            1.0         20.0   
1       3846729               20.0            2.0         40.0   
2       3847007               20.0            1.0         20.0   
3       3847801              100.0            1.0        100.0   
4       3848069               15.0            1.0         15.0   
...         ...                ...            ...          ...   
21474   5650343               50.0            2.0        100.0   
21475   5650798               50.0            3.0        100.0   
21476   5650844               25.0            1.0         25.0   
21477   5650849                0.0            0.0          0.0   
21478   5650876               25.0            1.0         25.0   

      Cash Outs Count Cash Outs Sum Bets Sum Wins Sum Bonus Issues Sum  \
                  sum           sum      sum      sum              sum   
0                 0.0           0.0    75.80    56.15             0.00   
1                 0.0           0.0     0.00     0.00             0.00   
2                 0.0           0.0    20.50     0.50             0.00   
3                 0.0           0.0   275.00    75.00           100.00   
4                 0.0           0.0   188.25   215.50             0.00   
...               ...           ...      ...      ...              ...   
21474             0.0           0.0   242.30   142.30             0.00   
21475             0.0           0.0   655.80   469.50            88.75   
21476             0.0           0.0    89.81   127.05             0.00   
21477             0.0           0.0     9.24     6.60             2.70   
21478             0.0           0.0     0.00     0.00            25.00   

      income_player Cashout_key wins_key  
                sum         sum      sum  
0            -19.65           0        0  
1              0.00           0        1  
2            -20.00           0        0  
3           -200.00           0        0  
4             27.25           0        1  
...             ...         ...      ...  
21474       -100.00           0        0  
21475       -186.30           0        0  
21476         37.24           0        1  
21477         -2.64           0        0  
21478          0.00           0        1  

[21479 rows x 12 columns]

In [534]:
player_report.columns = ['Player ID','FTD_sum','Deposits Count', 'Deposits Sum', 'Cash Outs Count', 'Cash Outs Sum','Bets Sum','Wins Sum','Bonus Issues Sum','Income_player','Cashout_days','Wins_days']

In [535]:
player_report.head()

Player ID  FTD_sum  Deposits Count  Deposits Sum  Cash Outs Count  \
0    3845721     20.0             1.0          20.0              0.0   
1    3846729     20.0             2.0          40.0              0.0   
2    3847007     20.0             1.0          20.0              0.0   
3    3847801    100.0             1.0         100.0              0.0   
4    3848069     15.0             1.0          15.0              0.0   

   Cash Outs Sum  Bets Sum  Wins Sum  Bonus Issues Sum  Income_player  \
0            0.0     75.80     56.15               0.0         -19.65   
1            0.0      0.00      0.00               0.0           0.00   
2            0.0     20.50      0.50               0.0         -20.00   
3            0.0    275.00     75.00             100.0        -200.00   
4            0.0    188.25    215.50               0.0          27.25   

   Cashout_days  Wins_days  
0             0          0  
1             0          1  
2             0          0  
3             0          0  
4             0          1

Теперь посмотрим средние по player_report, когда cashout был выше нуля

In [537]:
g = player_report.query('Cashout_days != 0').shape[0]
t = player_report.shape[0]
n = g / t

print('all players =',t)
print('all cashoup players =',g)
print('share of cashout players in total(%) =',round(n,2))

all players = 21479
all cashoup players = 1040
share of cashout players in total(%) = 0.05


In [538]:
player_report.query('Cashout_days > 0 and Wins_days > 0')['Cash Outs Sum'].sum()

2538522.88

Теперь выведем средние по игрокам которые не делают и делают кэшаут

In [542]:
player_report.query('Cashout_days == 0').mean().round(2)

Player ID           4834446.31
FTD_sum                  34.51
Deposits Count            3.71
Deposits Sum            203.07
Cash Outs Count           0.17
Cash Outs Sum             2.74
Bets Sum               3413.54
Wins Sum               3186.45
Bonus Issues Sum         48.67
Income_player          -227.09
Cashout_days              0.00
Wins_days                 0.32
dtype: float64

In [546]:
print('Средний инком перед кэшаутом =',df_filter_act.query('Cashout_key != 0 and income_player > 0')['income_player'].mean().round(2))

средний инком перед кэшаутом = 867.13


In [548]:
print('Средний убыток перед кэшаутом =',df_filter_act.query('Cashout_key != 0 and income_player < 0')['income_player'].mean().round(2))

Средний убыток перед кэшаутом = -724.09


In [540]:
player_report.query('Cashout_days != 0 ').mean().round(2)

Player ID           4631452.86
FTD_sum                  45.63
Deposits Count           37.75
Deposits Sum           3359.16
Cash Outs Count          11.48
Cash Outs Sum          2449.44
Bets Sum              54422.89
Wins Sum              53250.06
Bonus Issues Sum        255.69
Income_player         -1172.83
Cashout_days              3.07
Wins_days                 4.30
dtype: float64

In [550]:
print('Средний инком игрока перед кэшаутом =',df_filter_act.query('Cashout_key != 0 and income_player > 0')['income_player'].mean().round(2))

Средний инком игрока перед кэшаутом = 867.13


In [548]:
print('Средний убыток перед кэшаутом =',df_filter_act.query('Cashout_key != 0 and income_player < 0')['income_player'].mean().round(2))

Средний убыток перед кэшаутом = -724.09


Теперь посмотрим на показатели для разных сумм депов

In [557]:
player_report.query('Cashout_days != 0')['Deposits Sum'].describe().round(2)

count      1040.00
mean       3359.16
std        9160.58
min           0.03
25%         250.00
50%         800.00
75%        2826.25
max      203962.85
Name: Deposits Sum, dtype: float64

In [555]:
player_report.query('Cashout_days != 0 and `Deposits Sum` <= 500').mean().round(2)

Player ID           4714044.07
FTD_sum                  32.41
Deposits Count            6.08
Deposits Sum            199.56
Cash Outs Count           5.81
Cash Outs Sum           509.97
Bets Sum               6514.53
Wins Sum               6772.18
Bonus Issues Sum         50.14
Income_player           257.65
Cashout_days              1.29
Wins_days                 2.32
dtype: float64

In [553]:
player_report.query('Cashout_days != 0 and 500 < `Deposits Sum` <= 1000').mean().round(2)

Player ID           4715472.05
FTD_sum                  34.46
Deposits Count            9.35
Deposits Sum            347.59
Cash Outs Count           6.35
Cash Outs Sum           667.76
Bets Sum               9318.67
Wins Sum               9558.64
Bonus Issues Sum         69.44
Income_player           239.98
Cashout_days              1.53
Wins_days                 2.74
dtype: float64

In [556]:
player_report.query('Cashout_days != 0 and 1000 < `Deposits Sum` <= 2000').mean().round(2)

Player ID           4661444.36
FTD_sum                  52.04
Deposits Count           29.44
Deposits Sum           1437.59
Cash Outs Count           9.32
Cash Outs Sum          1442.17
Bets Sum              22000.27
Wins Sum              21867.80
Bonus Issues Sum        169.21
Income_player          -132.47
Cashout_days              2.51
Wins_days                 4.04
dtype: float64

In [566]:
player_report.query('Cashout_days != 0 and 2000 < `Deposits Sum` <= 3000').mean().round(2)

Player ID           4519851.36
FTD_sum                  41.60
Deposits Count           47.35
Deposits Sum           2529.00
Cash Outs Count          13.24
Cash Outs Sum          2545.25
Bets Sum              37129.72
Wins Sum              36992.01
Bonus Issues Sum        209.12
Income_player          -137.71
Cashout_days              3.82
Wins_days                 6.10
dtype: float64

In [583]:
player_report.query('Cashout_days != 0 and 3000 < `Deposits Sum` <= 5000').mean().round(2)

Player ID           4473275.74
FTD_sum                  56.31
Deposits Count           66.47
Deposits Sum           3807.01
Cash Outs Count          13.49
Cash Outs Sum          2702.87
Bets Sum              58511.84
Wins Sum              57211.22
Bonus Issues Sum        286.78
Income_player         -1300.63
Cashout_days              3.59
Wins_days                 4.69
dtype: float64

In [584]:
player_report.query('Cashout_days != 0 and 5000 < `Deposits Sum` <= 10000').mean().round(2)

Player ID           4458586.49
FTD_sum                  61.09
Deposits Count           92.12
Deposits Sum           7307.17
Cash Outs Count          20.89
Cash Outs Sum          5239.93
Bets Sum             129006.00
Wins Sum             126532.44
Bonus Issues Sum        572.59
Income_player         -2473.56
Cashout_days              6.67
Wins_days                 7.25
dtype: float64

In [585]:
player_report.query('Cashout_days != 0 and 10000 < `Deposits Sum`').mean().round(2)

Player ID           4437246.94
FTD_sum                  86.78
Deposits Count          156.39
Deposits Sum          23774.20
Cash Outs Count          37.69
Cash Outs Sum         13324.90
Bets Sum             358045.03
Wins Sum             345834.41
Bonus Issues Sum       1379.02
Income_player        -12210.63
Cashout_days              9.78
Wins_days                10.58
dtype: float64

In [581]:
print(' Deposits Sum <= 500:',player_report.query('Cashout_days != 0 and `Deposits Sum` < 500').shape[0])
print(' 500 < Deposits Sum <= 1000 :',player_report.query('Cashout_days != 0 and 500 < `Deposits Sum` <= 1000').shape[0])
print(' 1000 < Deposits Sum <= 2000 :',player_report.query('Cashout_days != 0 and 1000 < `Deposits Sum` <= 2000').shape[0])
print(' 2000 < Deposits Sum <= 3000 :',player_report.query('Cashout_days != 0 and 2000 < `Deposits Sum` <= 3000').shape[0])
print(' 3000 < Deposits Sum <= 5000 :',player_report.query('Cashout_days != 0 and 3000 < `Deposits Sum` <= 5000').shape[0])
print(' 5000 < Deposits Sum <= 10000 :',player_report.query('Cashout_days != 0 and 5000 < `Deposits Sum` <= 10000').shape[0])
print('10000 < Deposits Sum :',player_report.query('Cashout_days != 0 and 10000 < `Deposits Sum`').shape[0])
print(' Deposits Sum total <= 500:',player_report.query('Cashout_days != 0').shape[0])

 Deposits Sum <= 500: 398
 500 < Deposits Sum <= 1000 : 166
 1000 < Deposits Sum <= 2000 : 154
 2000 < Deposits Sum <= 3000 : 72
 3000 < Deposits Sum <= 5000 : 78
 5000 < Deposits Sum <= 10000 : 85
10000 < Deposits Sum : 83
 Deposits Sum total <= 500: 1040


## Общий Вывод
<a class="anchor" id="chapter6"></a>

Описание игрока который не делает кэшаут:

- 95%  доля таких игроков
- 0,3  победных дня в среднем на игрока без кэшаута
- 0    кэшаутов дня на в среднем на игрока
- 0    баксов (0) кэшауты
- 0    средний кэшаут
- 200  баксов cредняя сумма депов от игрока
- 4 депа cреднее число депов от игрока
- 50 баксов средний деп
- -220 баксов средний доход игрока
- 3,5к  средний оборот
- 50  средний бонус
- 35   средний FTD  по игроку

Описание игрока который делает кэшаут:

- 5% доля таких игроков
- 1,5 победных дня в среднем на игрока до кэшаута
- 11 кэшаут дня на в среднем на игрока
- 2400 баксов кэшауты
- 220 баксов средний кэшаут
- 3400 баксов cредняя сумма депов от игрока
- 38 депов cреднее число депов от игрока
- 90 баксов средний деп
- -1200 баксов средний доход игрока
- 55к средний оборот
- 250 средний бонус
- 45 средний FTD по игроку
- 75% игроков выводят в день когда выйгрыш больше или равен расходам
- средний выйгрыш перед выводом 850 доларов
- 25% выводят в день, когда выйгрыш меньше расходов игрока
- средний убыток перед выводом 700 долларов

- 55% игроков с суммой депов до 1'000 долларов, выводят в  +250 долларов
- 15% игроков с суммой депов от 1'000 до 2'000 долларов, выводят в +130 долларов
- 7% игроков с суммой депов от 2'000 до 3'000 долларов, выводят в -140 долларов
- 7% игроков с суммой депов от 3'000 до 5'000 долларов, выводят в -1'300 долларов
- 8% игроков с суммой депов от 5'000 до 10'000 долларов, выводят в -2'500 долларов
- 8% игроков с суммой депов от 10'000 долларов, выводят в -12'000 долларов